lizard_auth_client/mixins.py

```diff
@@ -0,0 +1,204 @@
+        if excluded_roles:
+            qs = qs.exclude(code__in=excluded_roles)
+
+        self._available_roles = qs
+        return self._available_roles
+
+    def get_role_required(self):
+        """
+        Override this method to override the role attribute.
+        Must return an iterable.
+        """
+        if not self.role_required:
+            raise ImproperlyConfigured(
+                '{0} is missing the role attribute. Define {0}.role, or '
+                'override {0}.get_role_required().'.format(
+                    self.__class__.__name__)
+            )
+        if isinstance(self.role_required, six.string_types):
+            roles = (self.role_required,)
+        else:
+            roles = self.role_required
+        return roles
+
+    def has_role(self):
+        """
+        Override this method to customize the way permissions are checked.
+        """
+        user = self.request.user
+        # superusers have all roles implicitly
+        if user.is_superuser:
+            return True
+
+        roles = self.get_role_required()
+        nr_of_user_organisation_roles = models.UserOrganisationRole.objects.\
+            filter(user=user, role__code__in=roles).count()
+        if nr_of_user_organisation_roles > 0:
+            return True
+        else:
+            return False
+
+    @method_decorator(login_required)
+    def dispatch(self, request, *args, **kwargs):
+        if not self.has_role():
+            return self.handle_no_permission()
+        return super(RoleRequiredMixin, self).dispatch(
+            request, *args, **kwargs)
+
+
+class ManagedObjectsMixin(object):
+    """
+    CBV mixin which provides a number of role/permission management-related
+    fields:
+
+    - managed_organisations: the organisations managed by the request user
+    - managed_users: the users managed by the request user for the given
+        organisation(s)
+
+    """
+    def get_managed_organisations(self):
+        """
+        Get the managed organisations for the current user. And set it as an
+        instance variable.
+        """
+        if hasattr(self, 'managed_organisations'):
+            return self.managed_organisations
+
+        # superusers are allowed to manage all objects
+        if self.request.user.is_superuser:
+            return models.Organisation.objects.all()
+
+        # fetch the managed organisations based on the UserOrganisationRole
+        # instances of the request user
+        user_organisation_roles = models.UserOrganisationRole.objects.filter(
+            user=self.request.user,
+            role__code__in=settings.SSO_MANAGER_ROLE_CODES)
+        self.managed_organisations = models.Organisation.objects.distinct().\
+            filter(user_organisation_roles__in=user_organisation_roles).\
+            order_by('name')
+        return self.managed_organisations
+
+    def get_managed_users(self, managed_organisations):
```

Comment:
Kan volgens mij hier niet, want get_managed_users verwacht een argument, die variabel is, namelijk managed_organisations.